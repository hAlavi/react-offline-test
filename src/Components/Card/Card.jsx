import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';


export const CardWrapper = styled.div` 
  display: flex;
  flex-wrap: wrap;
  align-items: stretch;
`;

const CardContainer = styled.div` 
  flex: 0 0 200px;
  width: 100%;
  margin: 10px;
  border: 1px solid #ccc;
  box-shadow: 2px 2px 6px 0px  rgba(0,0,0,0.3);
`;

const CardGauge = styled.div` 
  max-width: 100%;
`;

const CardTitle = styled.h3`
  width: 100%;
  font-size: 16px;
  text-align: center;
`; 

const Card = () => {
    return (
        <CardContainer>
            <CardGauge>THIS IS A PLACE HOLDER FOR GAUGE</CardGauge>
            <CardTitle>Gas</CardTitle>
        </CardContainer>
    );
};

export default Card;