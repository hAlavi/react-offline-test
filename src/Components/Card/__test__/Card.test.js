import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Card, { CardWrapper } from '../Card';
import 'jest-styled-components';

afterEach(cleanup);

describe('Unit tests for the Card & CardWrapper Component', () => {
  describe('Visual Snapshot testing', () => {
    it('Card - Renders without crashing', () => {
      const { container } = render(
        <Card title="title of card">Internal Children</Card>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('CardWrapper - Renders without crashing', () => {
      const { container } = render(
        <CardWrapper>
          <Card title="title1">Internal Children 1</Card>
          <Card title="title2">Internal Children 2</Card>
        </CardWrapper>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('Functional unit tests', () => {
    it('Card - it should have specific parts', () => {
      const { getByText } = render(
        <Card title="title of card">
          <div>Internal Children</div>
        </Card>,
      );

      expect(getByText('title of card')).toBeTruthy();
      expect(getByText('Internal Children')).toBeTruthy();
    });

    it('CardWrapper - it should have specific parts', () => {
      const { getByText } = render(
        <CardWrapper>
          <Card title="title1">
            <div>Internal Children 1</div>
          </Card>
          <Card title="title2">
            <div>Internal Children 2</div>
          </Card>
        </CardWrapper>,
      );

      expect(getByText('title1')).toBeTruthy();
      expect(getByText('Internal Children 1')).toBeTruthy();
      expect(getByText('title2')).toBeTruthy();
      expect(getByText('Internal Children 2')).toBeTruthy();
    });
  });
});
