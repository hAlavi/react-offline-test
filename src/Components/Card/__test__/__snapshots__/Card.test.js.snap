// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the Card & CardWrapper Component Visual Snapshot testing Card - Renders without crashing 1`] = `
.c0 {
  -webkit-flex: 0 0 200px;
  -ms-flex: 0 0 200px;
  flex: 0 0 200px;
  width: 100%;
  margin: 10px;
  border: 1px solid #ccc;
  box-shadow: 2px 2px 6px 0px rgba(0,0,0,0.3);
}

.c1 {
  max-width: 100%;
}

.c2 {
  width: 100%;
  font-size: 16px;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Internal Children
  </div>
  <h3
    class="c2"
  >
    title of card
  </h3>
</div>
`;

exports[`Unit tests for the Card & CardWrapper Component Visual Snapshot testing CardWrapper - Renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 {
  -webkit-flex: 0 0 200px;
  -ms-flex: 0 0 200px;
  flex: 0 0 200px;
  width: 100%;
  margin: 10px;
  border: 1px solid #ccc;
  box-shadow: 2px 2px 6px 0px rgba(0,0,0,0.3);
}

.c2 {
  max-width: 100%;
}

.c3 {
  width: 100%;
  font-size: 16px;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Internal Children 1
      </div>
      <h3
        class="c3"
      >
        title1
      </h3>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Internal Children 2
      </div>
      <h3
        class="c3"
      >
        title2
      </h3>
    </div>
  </div>
</div>
`;
